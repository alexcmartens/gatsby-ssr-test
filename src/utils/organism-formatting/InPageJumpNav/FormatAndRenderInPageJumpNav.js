import React from "react"
import { InPageJumpNav } from "terra-one"

const FormatAndRenderInPageJumpNav = ({ data, navTitle }) => {
  const displayedSections = []
  if (data && data !== null && Array.isArray(data)) {
    data?.forEach(section => {
      if (section?.anchor && section?.jumpNavTitle) {
        section.link = section?.anchor
        section.title = section?.jumpNavTitle
        displayedSections.push(section)
      }
    })
  }
  return (
    <>
      <InPageJumpNav
        pageSections={displayedSections}
        jumpNavTitle={navTitle && navTitle}
      />
    </>
  )
}

export default FormatAndRenderInPageJumpNav
