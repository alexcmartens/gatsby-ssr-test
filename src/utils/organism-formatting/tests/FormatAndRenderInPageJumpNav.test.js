import React from "react"
import renderer from "react-test-renderer"
import FormatAndRenderInPageJumpNav from "../InPageJumpNav/FormatAndRenderInPageJumpNav"

describe("FormatAndRenderInPageJumpNav", () => {
  it("renders correctly", () => {
    const pageSections = [
      { title: "Overview", link: "overview" },
      { title: "Uses by Industry", link: "uses" },
      { title: "Features & Benefits", link: "features" },
      { title: "Complete Solutions and Lorem Ipsum", link: "solutions" },
      { title: "Articles", link: "articles" },
    ]

    const tree = renderer
      .create(
        <FormatAndRenderInPageJumpNav
          pageSectionsData={pageSections}
          navTitle={"This Page"}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
