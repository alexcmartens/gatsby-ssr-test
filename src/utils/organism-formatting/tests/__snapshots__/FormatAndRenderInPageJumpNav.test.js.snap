// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormatAndRenderInPageJumpNav renders correctly 1`] = `<div />`;
